lib/vnet/daemon/client_darwin.m

```diff
@@ -0,0 +1,99 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#import <Foundation/Foundation.h>
+#import <ServiceManagement/ServiceManagement.h>
+
+#include "client_darwin.h"
+
+void BundlePath(struct BundlePathResult *result) {
+    // From the docs:
+    // > This method may return a valid bundle object even for unbundled apps.
+    // > It may also return nil if the bundle object could not be created,
+    // > so always check the return value.
+    NSBundle *main = [NSBundle mainBundle];
+    if (!main) {
+        result->bundlePath = strdup("");
+        return;
+    }
+    
+    result->bundlePath = VNECopyNSString([main bundlePath]);
+}
+
+NSString * DaemonLabel(void) {
+    NSBundle *main = [NSBundle mainBundle];
+    if (!main) {
+        return @"";
+    }
+    
+    NSString *bundleIdentifier = [main bundleIdentifier];
+    if ([bundleIdentifier length] == 0) {
+        return bundleIdentifier;
+    }
+    
+    return [NSString stringWithFormat:@"%@.vnetd", bundleIdentifier];
```

Comment:
Could we get away with the app not knowing its own bundle? I think that would be best, if we can.

Reply:
I'll think about it, but off the top of my head I think that won't be possible. The daemon needs to have a label that's advertised in the plist _and_ we want to use different labels between tshdev and tsh, just as we do with the bundle identifier of tsh.app.

I wanted to avoid injecting the same identifier twice, e.g. once in plist and another time through build tags just so that `client_darwin.go` get the identifier without consulting `[[NSBundle mainBundle] bundleIdentifier]`.

Reply:
If we need it in the bundle then querying NSBundle seems for the best. +1.